pkg/testutil/require.go

```diff
@@ -228,23 +235,57 @@ func testPodsWithPredicate(pods []corev1.Pod, pred func(*corev1.Pod) bool) bool
 	return true
 }
 
-func statefulSetIsReady(ss *appsv1.StatefulSet) bool {
-	return ss.Status.ReadyReplicas == ss.Status.Replicas
-}
+func statefulSetIsReady(t *testing.T, sts *appsv1.StatefulSet, sb testenv.DiffingSandbox) (bool, error) {
+	if sts.Status.ReadyReplicas != sts.Status.Replicas {
```

Comment:
I think you might want an additional check of `.Generation > .Status.ObservedGeneration` here? It might be possible that you're picking up old pods?